### PR TITLE
Custom models and datasets

### PR DESCRIPTION
**Pull request by jxmorris12**
- `--model-from-file` and `--dataset-from-file` to load models & datasets from a file
- `--model-from-huggingface` loads models from [huggingface pretrained models](https://huggingface.co/models)
- `--dataset-from-nlp` loads datasets from [`nlp` library](https://github.com/huggingface/nlp)
- `--attack-from-file`
- tests that cover each thing
- new `bert-base-uncased-mnli` model replaces the old one
- behind the scenes: completely redo tokenization 
- add some info to the readme

minor changes:
- refactor utils to different files
- update `benchmark_models.py` script
- `textattack.shared.utils.get_logger() -> textattack.shared.logger`
- `textattack.shared.utils.get_device() -> textattack.shared.utils.device`
- rename `run_attack_args_helper` to `attack_args_helper`

### CODE REVIEW

**Comment by jxmorris12** [comment]
I realized it would help a lot to have "shortcuts" to pairs of huggingface models & datasets, like we do for our pretrained datasets. For example, `--model distilbert-sst2` should automatically select the pretrained model and dataset.

File: textattack/shared/scripts/attack_args_helper.py [hunk 1]
```diff
@@ -353,12 +422,12 @@ def parse_goal_function_and_attack_from_args(args):
...
 
-def parse_logger_from_args(args):# Create logger
+def parse_logger_from_args(args):
+    # Create logger
     attack_log_manager = textattack.loggers.AttackLogManager()
     # Set default output directory to `textattack/outputs`.
     if not args.out_dir:
-        current_dir = os.path.dirname(os.path.realpath(__file__))
-        outputs_dir = os.path.join(current_dir, os.pardir, os.pardir, os.pardir, 'outputs')
+        outputs_dir = os.path.join(os.getcwd(), 'textattack_outputs')
```
Comment (jinyongyoo) [inline textattack/shared/scripts/attack_args_helper.py:430]: I think the default behavior right now for non-existent output directory is to create it. I'm worried that if the user is not in the right current working directory, we'll end up confusing the users as to where they placed the outputs. I think for default outputs, we should at least be more consistent than cwd.

File: textattack/datasets/huggingface_nlp_dataset.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
+from textattack.datasets import TextAttackDataset
+from textattack.shared import TokenizedText
+
+import nlp
+import random
+
+def get_nlp_dataset_columns(dataset):
```
Comment (uvafan) [inline textattack/datasets/huggingface_nlp_dataset.py:7]: Am I missing something? Why are all the outputs "labels"? Do we supporot seq2seq datasets?

File: textattack/shared/scripts/attack_args_helper.py [hunk 3]
```diff
@@ -202,7 +270,7 @@ def get_args():
         help='Run attack using multiple GPUs.')
 
     goal_function_choices = ', '.join(GOAL_FUNCTION_CLASS_NAMES.keys())
-    parser.add_argument('--goal-function', '-g', default='untargeted-classification',
+    parser.add_argument('--goal-function', default='untargeted-classification',
```
Comment (uvafan) [inline textattack/shared/scripts/attack_args_helper.py:273]: why are you getting rid of the `-g` type options?

File: README.md [hunk 4]
```diff
@@ -85,23 +115,71 @@ of a string or a list of strings. Here's an example of how to use the `Embedding
...
+TextAttack is model-agnostic! You can use `TextAttack` to analyze any model that outputs IDs, tensors, or strings.
+
+#### HuggingFace `transformers` and `nlp`
+
+We now provide built-in support for [`transformers` pretrained models](https://huggingface.co/models) 
+and datasets from the [`nlp` package](https://github.com/huggingface/nlp)! Here's an example of loading
+and attacking a pre-trained model and dataset:
+
+```
+python -m textattack --model_from_huggingface distilbert-base-uncased-finetuned-sst-2-english --dataset_from_nlp glue:sst2 --recipe deepwordbug --num-examples 10
```
Comment (uvafan) [inline README.md:132]: \- instead of _

File: README.md [hunk 5]
```diff
@@ -85,23 +115,71 @@ of a string or a list of strings. Here's an example of how to use the `Embedding
...
+```
+
+You can explore other pre-trained models using the `--model_from_huggingface` argument, or other datasets by changing
```
Comment (uvafan) [inline README.md:135]: \- instead of _

File: README.md [hunk 6]
```diff
@@ -85,23 +115,71 @@ of a string or a list of strings. Here's an example of how to use the `Embedding
...
+##### Model from a file
+, you could create the following file
+and name it `my_model.py`:
+
+```
+model = load_model()
+tokenizer = load_tokenizer()
+```
+
+Then, run an attack with the argument `--model_from_file my_model.py`. The model and tokenizer will be loaded automatically.
```
Comment (uvafan) [inline README.md:155]: \- instead of _

File: local_tests/command_line_tests.py [hunk 7]
```diff
@@ -26,6 +26,26 @@ def register_test(command, name=None, output_file=None, desc=None):
     output_file='local_tests/sample_outputs/interactive_mode.txt', 
     desc='Runs textfooler attack on BERT trained on IMDB using interactive mode')
 
+#
+# test: run_attack on HuggingFace ``transformers`` model & ``nlp`` dataset
+#
+register_test(('python -m textattack --model_from_huggingface '
```
Comment (uvafan) [inline local_tests/command_line_tests.py:32]: \- instead of _

File: local_tests/command_line_tests.py [hunk 8]
```diff
@@ -26,6 +26,26 @@ def register_test(command, name=None, output_file=None, desc=None):
...
+    'distilbert-base-uncased-finetuned-sst-2-english '
+    '--dataset_from_nlp glue:sst2 --recipe deepwordbug --num-examples 5'),
+    name='huggingface_model_dataset',
+    output_file='local_tests/sample_outputs/run_attack_transformers_nlp.txt',
+    desc='Runs DeepWordBug recipe against pre-trained model and dataset')
+    
+#
+# test: run_attack on model and dataset loaded from file
+#
+register_test(('python -m textattack --model_from_file local_tests/sample_inputs/sst_model_and_dataset.py '
```
Comment (uvafan) [inline local_tests/command_line_tests.py:42]: \- instead of _

File: textattack/shared/scripts/attack_args_helper.py [hunk 9]
```diff
@@ -358,7 +428,89 @@ def parse_attack_from_args(args):
...
+            tokenizer = getattr(model_module, tokenizer_name)
+        except AttributeError:
+            raise AttributeError(f'``{tokenizer_name}`` not found in module {args.model_from_file}')
+        model = model.to(textattack.shared.utils.device)
+        setattr(model, 'tokenizer', tokenizer)
+    elif args.model_from_huggingface:
+        import transformers
+        colored_model_name = textattack.shared.utils.color_text(args.model_from_huggingface, color='blue', method='ansi')
+        textattack.shared.logger.info(f'Loading pre-trained model from HuggingFace model repository: {colored_model_name}')
+        model = transformers.AutoModelForSequenceClassification.from_pretrained(args.model_from_huggingface)
```
[the remark this answers is not in the document]
Comment (jxmorris12) [reply textattack/shared/scripts/attack_args_helper.py:458]: thanks, it should actually just be `AutoModel`
